### ISSUE TITLE
Unexplained "Error checking for update"

### ISSUE THREAD

**Issue by mehov** [greasemonkey/greasemonkey#3182]
Hi everyone and thank you for making Greasemonkey! I have been a Firefox user for nearly two decades, and heard of Greasemonkey many many times, but never actually used it – until now. Feels like a great tool to have!

I wrote a simple "hello world" script:

```
// ==UserScript==
// @name    Hello
// @version 1
// @grant   none
// ==UserScript==

alert('hello greasemonkey');
```

It works fine, but does not update. When I go to _plugins_ → _Greasemonkey_ → _User scripts for this tab_ → _User script options_ → _Update now_, I'm getting *Error checking for update*.

- The error appears under the Update button immediately, as if no network request to my actual script is even attempted
- Nothing is reported in the browser console or in the network requests debugger
- This also happens also if I have `@updateURL` and `@downloadURL` in the script

Is there a place where I can see a more verbose output maybe? Or can the error be made more specific?

**Comment by makyen** [comment]
What you've provided implies that you wrote the userscript in the Greasemonkey editor, rather than installing it from a URL, so Greasemonkey wouldn't have been able to get the URL during the install process. The userscript header you've shown here has nothing defined which would provide any clue to Greasemonkey as to where it should be looking for updates to the userscript, so, of course, updates will fail.

You haven't provided what you actually put in the userscript for `@updateURL` or `@downloadURL`, so there's no way for us to say anything as to if what you used should have worked.

Given the general confusion that's involved with looking at output from browser extensions, it's likely you're looking in the wrong place for such output. I haven't reread them recently, but my answers on Stack Overflow: [answer to: "How do I see the console.log output of a background script in a Firefox WebExtension?"](https://stackoverflow.com/a/39084633/3773011) and [answer to: "How to open the correct devtools console to see output from an extension script?"](https://stackoverflow.com/a/38920982/3773011) should at least be pointing in the right general direction for how to view an appropriate console.

**Comment by Cerberus-tm** [comment]
> Given the general confusion that's involved with looking at output from browser extensions, it's likely you're looking in the wrong place for such output. I haven't reread them recently, but my answers on Stack Overflow: [answer to: "How do I see the console.log output of a background script in a Firefox WebExtension?"](https://stackoverflow.com/a/39084633/3773011) and [answer to: "How to open the correct devtools console to see output from an extension script?"](https://stackoverflow.com/a/38920982/3773011) should at least be pointing in the right general direction for how to view an appropriate console.

From your post there:

> The [Browser Console](https://developer.mozilla.org/en-US/docs/Tools/Browser_Console) no longer shows output from WebExtensions background pages by default. You can have it show output from all WebExtensions by selecting to display "Show Content Messages", which is available from the popup that opens when you click on the gear-like symbol "⚙️" in the upper right of the window, just to the right of "Requests".

But I do not see "Show Content Messages", is that out of date?

**Comment by makyen** [comment]
@Cerberus-tm Yes, that appears to be outdated. They seem to have removed that option and added a "Browser Console Mode" banner where you can select "Parent process only (Fast)" or "Multiprocess (Slower)". Unfortunately, they also appear to have broken it in that it refuses to not scroll with newly added information, even if you've manually scrolled back to some point in time (which should, but doesn't, remain static in the display at the point you've scrolled back to). In "Multiprocess (Slower)" mode it also appears to be a concatenation of all Web Consoles, which is largely unusable due to the scrolling issue (i.e., in an actively used profile, I wasn't able to even scroll back to look at something specific without it scrolling off the screen). In other words, in that mode on a profile with many tabs, it's a firehose of information which you can't stop and look at a single point (which you used to be able to do). That mode of the Browser Console could still be useful for some things, particularly in a test profile, but not if there are multiple things going on in an actively used profile under normal conditions.

So, for a single extension's background pages you're better off by focusing just on that extension's pages, which can be inspected through `about:debugging`.

**Comment by Cerberus-tm** [comment]
@makyen Actially, I can scroll back just fine, and it stayed in the scrolled position even if new lines appear at the bottom. So I wonder what is going on in your profile.

About "Show Content Messages": is there now no way to show those? Or are they always shown now?

**Comment by mehov** [comment]
> What you've provided implies that you wrote the userscript in the Greasemonkey editor, rather than installing it from a URL

I am sorry for the confusion, I actually hosted it on a website having a `.user.js` public URL and installed it from there
